### DISCUSSION TITLE
[V5] How to get Host name in model?

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#2634] [Help]
Hi,
I need host name in model, how I get this? @thetutlage

**Comment by pk4all** [comment]
Any suggestion pls @RomainLanz

**Comment by RomainLanz** [comment]
Hey @pk4all! 👋 

Please, be more descriptive in your question.
There are no way we can help you with the amount of information you shared.

**Reply by pk4all** [reply]
@RomainLanz I have a modal User and I want to return a field URL as host name on User modal call.
I mean if I fetch record from abc.com return abc.com as URL and if I fetch record from xyz.com return xyz.com as URL.

**Reply by RomainLanz** [reply]
It is still not clear.
What do you mean by fetching a record from `abc.com` or `xyz.com`?

**Reply by pk4all** [reply]
Hi @RomainLanz ,
Sorry for late reply actually in my app I have declare  a computed field 
```javascript
@computed()
  public get profileUrl () {
    const uid:any = this?.id
    //return uid
    var b = Buffer.from(''+uid)
    var s = b.toString('base64')
    return host+'/expert-profile/'+s
  }
```
I want to ```host``` variable according to host I mean on local "http://localhost" and on live "http://live.com"
Currently I have declare host in .env file.

Thanks

**Comment by Melchyore** [comment]
You can access to your [environment variables](https://docs.adonisjs.com/guides/environment-variables).

**Comment by amir-mohammad-HP** [comment]
there is no direct approach to this , if you are have a multi-site system using same code base , you have to pass the request's host directly to your method when is called 

```typescript
  public profileUrl (host: string) {
    const uid:any = this?.id
    //return uid
    var b = Buffer.from(''+uid)
    var s = b.toString('base64')
    return host+'/expert-profile/'+s
  }
```
and serialized it using serialize extras ( i'm not sure how it is in version 5

--- 
you may also consider using async local storage to access request's host [ALS](https://v5-docs.adonisjs.com/guides/async-local-storage#-works-1)
